src/main/java/org/gradle/profiler/CommandExec.java

```diff
@@ -248,10 +254,30 @@ public void interrupt() {
         }
 
         public void kill() {
+            // On Windows subprocesses are not automatically
+            // killed with parent, so let's do that manually
+            destroyDescendants();
             process.destroy();
             shutdownExecutor();
         }
 
+        private void destroyDescendants() {
+            if (!JavaVersion.current().isJava9Compatible()) {
+                // ProcessHandle API is available only from JDK9
+                return;
+            }
+            try {
+                @SuppressWarnings("unchecked")
+                Stream<Object> stream = (Stream<Object>) Process.class.getMethod("descendants").invoke(process);
```

Comment:
❓ Can we move this code to some helper class that is compiled against Java 9+, so we don't need to use reflection?

Reply:
I think moving this to a separate project and compiling it against Java9+ adds quite some complication. I would not do that for these 2 method calls.

Reply:
Aren't we running the build with Java 11? If not then let's leave it as is.

Reply:
Not all all platforms. On Mac and Linux we still run with Java8. We do that so we have at least some coverage with Java8, that we still support